Commit dialog honors value of `status.showUntrackedFiles'  git config.

mlutescu opened the pull request:
Hello, 

This is first time when I use GitHub to contribute to somebody's else project, also I have practically 0 experience with C#; so please forgive me if I do something patently wrong.  My small contribution tries to add feature described in https://github.com/gitextensions/gitextensions/issues/1872.

KindDragon commented on GitUI/CommandsDialogs/FormCommit.cs:
```diff
@@ -229,6 +229,7 @@ private void FormCommit_Load(object sender, EventArgs e)
             ResetUnStaged.Visible = AppSettings.ShowResetUnstagedChanges;
             CommitAndPush.Visible = AppSettings.ShowCommitAndPush;
             AdjustCommitButtonPanelHeight();
+	    showUntrackedFilesToolStripMenuItem.Checked = !Module.RunGitCmd("config status.showUntrackedFiles").Trim().Equals("no");
```
You should use Module.EffectiveConfigFile instead and we use 4 space indentation.

mlutescu commented:
Hello Arkady,

Thanks for reviewing my commit and for the advise.  I implemented your suggestion and I use Module.EffectiveConfigFile.  Also I corrected the spacing problem.

It looks like I get into a bug for the mechanism that Module.EffectiveConfigFile uses to read the settings: it does not cope well with multiple occurrences of the same section, a situation which someone may get into if <em>.git/config</em> is manipulated via <code>git config</code> as I described in my second commit comment.  Please advise.

mlutescu commented:
Should I close this pull request an make a new one?

KindDragon commented on GitUI/CommandsDialogs/FormCommit.cs:
```diff
@@ -229,7 +229,7 @@ private void FormCommit_Load(object sender, EventArgs e)
             CommitAndPush.Visible = AppSettings.ShowCommitAndPush;
             AdjustCommitButtonPanelHeight();
-	    showUntrackedFilesToolStripMenuItem.Checked = !Module.RunGitCmd("config status.showUntrackedFiles").Trim().Equals("no");
+            showUntrackedFilesToolStripMenuItem.Checked = !Module.EffectiveConfigFile.GetValue("status.showUntrackedFiles").Equals("no");
```
Better use string.Equals(val, "astringvalue", StringComparison.OrdinalIgnoreCase) to ignore case

KindDragon commented:
> Should I close this pull request an make a new one?

Just squash commits and force push to your branch

KindDragon commented:
> It looks like I get into a bug for the mechanism that Module.EffectiveConfigFile uses to read the settings: it does not cope well with multiple occurrences of the same section, a situation which someone may get into if .git/config is manipulated via git config as I described in my second commit comment. Please advise.

Please create separate bug about that if you can't fix it currently
